### PR TITLE
update disk-pool-volume-create --mount-option to only accept one para…

### PR DESCRIPTION
**Pull request by jonathanhe-msft**
Bugfix for https://github.com/Azure/azure-cli-extensions/issues/4395

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/vmware/azext_vmware/tests/latest/recordings/test_vmware_datastores.yaml [hunk 1]
```diff
@@ -126,7 +126,44 @@ interactions:
...
-      - Thu, 18 Nov 2021 21:47:50 GMT
+      - Tue, 08 Feb 2022 17:29:03 GMT
+      server:
+      - Rocket
+    status:
+      code: 200
+      message: OK
+- request:
+    body: '{"properties": {"diskPoolVolume": {"targetId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rasivagu-df-rg/providers/Microsoft.StoragePool/diskPools/rasivagu-df-diskpool/iscsiTargets/rasivagu-df-target",
+      "lunName": "lun0", "mountOption": "ATTACH"}}}'
```
Comment (cataggar) [inline src/vmware/azext_vmware/tests/latest/recordings/test_vmware_datastores.yaml:137]: Looks good 👍

**Review by cataggar — APPROVED** [review]
Thanks for the bug fix and test.

**Comment by yonzhan** [comment]
vmware

File: src/vmware/azext_vmware/tests/latest/test_datastores_scenario.py [hunk 2]
```diff
@@ -48,3 +48,6 @@ def test_vmware_datastores(self):
 
         # Delete the newly created ANF based datastore
         self.cmd('az vmware datastore delete --name iSCSIDatastore1 --resource-group {rg} --private-cloud {privatecloud} --cluster {cluster} --yes')
+
+        # Create a new iSCSI based datastore with --mount-option as ATTACH
+        self.cmd('az vmware datastore disk-pool-volume create --name iSCSIDatastore1 --resource-group {rg} --private-cloud {privatecloud} --cluster {cluster} --target-id {target_id} --lun-name lun0 --mount-option ATTACH')
```
Comment (wangzelin007) [inline src/vmware/azext_vmware/tests/latest/test_datastores_scenario.py:53]: Please add new line at the end of file.

File: src/vmware/azext_vmware/_params.py [hunk 3]
```diff
@@ -125,7 +125,7 @@ def load_arguments(self, _):
 
     with self.argument_context('vmware datastore disk-pool-volume create') as c:
         c.argument('target_id', help='Azure resource ID of the iSCSI target.')
-        c.argument('mount_option', nargs='*', help='Mode that describes whether the LUN has to be mounted as a datastore or attached as a LUN.')
+        c.argument('mount_option', help='Mode that describes whether the LUN has to be mounted as a datastore or attached as a LUN. Possible values include: "MOUNT", "ATTACH". Default value: "MOUNT".')
```
Comment (wangzelin007) [inline src/vmware/azext_vmware/_params.py:128]:
```suggestion
        c.argument('mount_option', arg_type=get_enum_type(['MOUNT', 'ATTACH']), default="MOUNT", help='Mode that describes whether the LUN has to be mounted as a datastore or attached as a LUN.')

```

**Review by wangzelin007 — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**